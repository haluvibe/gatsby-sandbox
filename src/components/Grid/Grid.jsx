import React from 'react';
import Column from '../Column/Column';
import uuidv4 from 'uuid/v4';

export default ({columns, settings}) => {
  const {className, ...other} = settings;
  return (
    <div className={`${className} l-grid`} {...other}>
      { columns && columns.map(component => <Column key={uuidv4()} {...component} />) }
    </div>
  )
}
