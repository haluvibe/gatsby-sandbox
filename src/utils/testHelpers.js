export const mockSettings = {
  id: 'id',
  className: 'a-class another-class',
  style: {
    backgroundColor: 'red',
    fontSize: 16,
  },
};

export const mockChildren = [
  {
    settings: mockSettings,
    component: 'section',
  },
];

export const mockColumns = [
  {
    settings: mockSettings,
    component: 'column',
  },
];

export const mockData = {
  site: {
    siteMetadata: {
      title: 'Mock Title',
    },
  },
};

export const mockHtml = '<div>Hello test world</div>';
