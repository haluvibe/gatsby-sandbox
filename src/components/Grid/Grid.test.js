import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Grid from './Grid';
import { mockSettings, mockColumns } from '../../utils/testHelpers';
jest.mock('uuid/v4', () => jest.fn(() => '9999-9999'));

test('Grid component should render as expected', () => {
  const component = shallow(
    <Grid settings={mockSettings} columns={mockColumns} />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
