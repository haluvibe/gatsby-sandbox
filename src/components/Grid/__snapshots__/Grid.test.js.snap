// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should render as expected 1`] = `
<div
  className="a-class another-class l-grid"
  id="id"
  style={
    Object {
      "backgroundColor": "red",
      "fontSize": 16,
    }
  }
>
  <Component
    component="column"
    key="9999-9999"
    settings={
      Object {
        "className": "a-class another-class",
        "id": "id",
        "style": Object {
          "backgroundColor": "red",
          "fontSize": 16,
        },
      }
    }
  />
</div>
`;
